js/tuxexchange.js

```diff
@@ -0,0 +1,567 @@
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchTicker () requires a "symbol" argument');
+        }
+        let res = await this.publicGet ({ 'method': 'getticker' });
+        const tickersData = this.parseJson (res);
+        const tickerIds = Object.keys (tickersData);
+        for (let i = 0; i < tickerIds.length; i++) {
+            const id = tickerIds[i];
+            const tickerResult = this.parseTicker (id, tickersData[id]);
+            const tickerSymbol = tickerResult['symbol'];
+            if (tickerSymbol === symbol) {
+                return tickerResult;
+            }
+        }
+        throw new BadRequest (this.id + ' no ticker found for symbol provided');
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        let res = await this.publicGet ({ 'method': 'getticker' });
+        const tickersData = this.parseJson (res);
+        const tickerIds = Object.keys (tickersData);
+        const result = {};
+        for (let i = 0; i < tickerIds.length; i++) {
+            const id = tickerIds[i];
+            const tickerResult = this.parseTicker (id, tickersData[id]);
+            const symbol = tickerResult['symbol'];
+            result[symbol] = tickerResult;
+        }
+        return result;
+    }
+
+    parseTicker (marketId, tickerData) {
+        const timestamp = this.nonce (); // TODO: hopefully this approximation is sufficient
+        const market = this.findMarket (marketId);
+        // ALGO: all the values here check out, but there are a lot of undefineds
+        return {
+            'symbol': market['symbol'],
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'close': this.safeFloat (tickerData, 'last'),
+            'last': this.safeFloat (tickerData, 'last'),
+            'high': this.safeFloat (tickerData, 'high24hr'),
+            'low': this.safeFloat (tickerData, 'low24hr'),
+            'percentage': this.safeFloat (tickerData, 'percentChange'),
+            'baseVolume': this.safeFloat (tickerData, 'baseVolume'),
+            'quoteVolume': this.safeFloat (tickerData, 'quoteVolume'),
+            'bid': this.safeFloat (tickerData, 'highestBid'),
+            'ask': this.safeFloat (tickerData, 'lowestAsk'),
+            'bidVolume': undefined,
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'previousClose': undefined,
+            'change': undefined,
+            'average': undefined,
+            'info': tickerData,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrderBook () requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const codes = this.getIdsFromSymbol (symbol);
```

Comment:
Alternatively, you can call `market` for the symbol which will do the supported check for you and then build the request based on the quote currency in the market

Reply:
that seems roundabout: I have a string with the two pieces of info that I need. If you're sure I"ll change it, otherwise I can flag it in the review and ask the maintainers

Reply:
whichever you prefer, just wanted to flag that up to you